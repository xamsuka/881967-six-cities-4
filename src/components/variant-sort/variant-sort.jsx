import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {SortTypes} from '../../const.js';
import {ActionCreator} from '../../reducers/application/reducer.js';
import {getCurrentSort} from '../../reducers/application/selectors.js';

class VariantSort extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {currentSort, onSortTypeChange, isOpen, onSortBlockClick} = this.props;
    const sortTypes = Object.values(SortTypes);
    const classSort = isOpen ? `places__options--opened` : ``;

    return (
      <form className="places__sorting" action="#" method="get">
        <span className="places__sorting-caption">Sort by </span>
        <span className="places__sorting-type" tabIndex={0} onClick={onSortBlockClick}>
          {currentSort}
          <svg className="places__sorting-arrow" width={7} height={4}>
            <use xlinkHref="#icon-arrow-select" />
          </svg>
        </span>
        <ul className={`places__options places__options--custom ${classSort}`} onClick={onSortTypeChange}>
          {sortTypes.map((sortType) => {
            const classActive = currentSort === sortType ? ` places__option--active` : ``;
            return (
              <li className={`places__option${classActive}`} tabIndex={0} key={sortType}>
                {sortType}
              </li>
            );
          })}
        </ul>
      </form>
    );
  }
}

VariantSort.propTypes = {
  currentSort: PropTypes.oneOf([`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`]).isRequired,
  onSortTypeChange: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onSortBlockClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  currentSort: getCurrentSort(state),
});

const mapDispatchToProps = (dispatch) => ({
  onSortTypeChange: (evt) => {
    dispatch(ActionCreator.changeSortType(evt));
  },
});

export {VariantSort};
export default connect(mapStateToProps, mapDispatchToProps)(VariantSort);
