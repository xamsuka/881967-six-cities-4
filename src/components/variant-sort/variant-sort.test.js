import React from 'react';
import renderer from 'react-test-renderer';
import {VariantSort} from './variant-sort.jsx';

test(`VariantSort render`, () => {
  const three = renderer
    .create(<VariantSort currentSort = {`Popular`} onSortTypeChange = {() => {}} />).toJSON();

  expect(three).toMatchSnapshot();
});
