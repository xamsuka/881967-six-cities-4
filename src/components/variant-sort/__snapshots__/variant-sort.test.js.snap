// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantSort render 1`] = `
<ul
  className="places__options places__options--custom places__options--opened"
  onClick={[Function]}
>
  <li
    className="places__option places__option--active"
    tabIndex={0}
  >
    Popular
  </li>
  <li
    className="places__option"
    tabIndex={0}
  >
    Price: low to high
  </li>
  <li
    className="places__option"
    tabIndex={0}
  >
    Price: high to low
  </li>
  <li
    className="places__option"
    tabIndex={0}
  >
    Top rated first
  </li>
</ul>
`;
